src/main/java/net/dv8tion/jda/core/requests/IGatewayProviderFactory.java

```diff
@@ -0,0 +1,25 @@
+package net.dv8tion.jda.core.requests;
```

Comment:
Put this and the other interface in a separate package "...requests.factory" together with a package-info.java